rasa/core/actions/forms.py

```diff
@@ -165,16 +166,44 @@ def _entity_mapping_is_unique(
         mapping_as_string = json.dumps(slot_mapping, sort_keys=True)
         return mapping_as_string in self._unique_entity_mappings
 
-    @staticmethod
+    def get_ignored_intents(self, domain: Domain) -> List[Text]:
+        """Returns a list of ignored intents as retrieved from `ignored_intents`.
+
+        Args:
+            domain: The current model domain.
+
+        Returns:
+            The value/s found in `ignored_intents` parameter in the `domain.yml`
+            (under forms).
+        """
+        if domain.forms:
```

Comment:
why do we need the `if domain.forms` check?

Reply:
Otherwise, this test `pytest tests/core/actions/test_forms.py::test_extract_requested_slot_default` fails because `domain.forms` is an empty dict `{}`.

Reply:
In that case the test `test_extract_requested_slot_default` needs adaption as we shouldn't use an empty domain but actually one which specifies the slot mappings for the form.

Reply:
> It is actually a domain that has an empty form like shown below :

You can see that in the implementation of `Domain.empty` [here](https://github.com/RasaHQ/rasa/blob/3e7212c1f6e95f518e2749c576676559b1dae756/rasa/shared/core/domain.py#L124)

> We should not keep the test case of empty form since there might be a user who puts an empty form in a domain?

if `domain.forms` is empty then there is also no way that we'd call this form in the first place. 

> as we see an empty form

Note that `if domain.forms` doesn't test if a single form is empty but only whether there are **any** forms given in the domain.